extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java

```diff
@@ -107,6 +109,8 @@ SyntheticBeanBuildItem initBasicAuth(
             configurator.defaultBean();
         }
 
+        securityInformationProducer.produce(SecurityInformationBuildItem.BASIC());
```

Comment:
This line should be conditional on `buildTimeConfig.auth.basic` (since `BasicAuthenticationMechanism` is only registered in `setupAuthenticationMechanisms` if the condition is true) and also inside the block above (where `configurator.defaultBean` is - where it is a default)